From 28109b9699dccc8199d116df15e0120b16ff5551 Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date: Thu, 4 Apr 2013 00:14:09 +0200
Subject: [PATCH 432/675] drm/rcar-du: Take offsets into account in the
 mode_set_base handler

Store the x and y offsets passed to the mode_set_base handler in the
plane structure. The offsets will be programmed to the hardware when
updating the plane base address.

Signed-off-by: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
(cherry picked from commit e2d54e4908bf4a2c3ecd06e702e18269cae78ecf)

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
---
 drivers/gpu/drm/rcar-du/rcar_du_crtc.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
index 08a69db..01e9d24 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_crtc.c
@@ -445,6 +445,11 @@ static void rcar_du_crtc_mode_commit(struct drm_crtc *crtc)
 static int rcar_du_crtc_mode_set_base(struct drm_crtc *crtc, int x, int y,
 				      struct drm_framebuffer *old_fb)
 {
+	struct rcar_du_crtc *rcrtc = to_rcar_crtc(crtc);
+
+	rcrtc->plane->src_x = x;
+	rcrtc->plane->src_y = y;
+
 	rcar_du_crtc_update_base(to_rcar_crtc(crtc));
 
 	return 0;
-- 
1.8.2

